### ISSUE TITLE
Group submission

### ISSUE THREAD

**Issue by MarieAugerMethe** [dakep/stat305templates#1]
I wanted to transform some of the learnr labs so group submit one lab sheet together rather than each individual submitting lab sheets. I thought it would be easy, but realize that because of the dependence of submit_lab_btn() on lab_name_input() it’s a bit more complicated than I thought. Any chance you could help. Any tip would be super appreciated!

**Comment by MarieAugerMethe** [comment]
FYI, I found I quick fix, that is a bit ugly, but should work. If you have a good fix however, I'm still keen to know.

**Comment by dakep** [comment]
Thanks for the suggestion! I updated the lab format in commit 59c70ec so that the *student_id* input is now optional. It can be disabled by setting `lab_name_input(label_id = NULL, ...)`.

For a group lab, I would suggest the following change the code chunk


````md
```{r}
# MUST BE PRESENT: the input for the student's name.
lab_name_input()
```
````

to something like

````md
```{r}
# MUST BE PRESENT: the input for the student's name.
lab_name_input(title = "Group information",
               label = "Group members",
               error_name_empty = "Please provide the names of all group members present today.",
               label_id = NULL)
```
````

By setting `label_id = NULL`, the second input (student nr) will not be shown.

Let me know if there's an issue with this update.

**Comment by MarieAugerMethe** [comment]
Thank you so much David. I really appreciate. One quick comment. Now the warning doesn't come up when you you use the old method. By which I mean, when you do individual submissions, rather than group submission. I'm guessing that for compatibility, it would be good if it have the same behaviour.

**Comment by MarieAugerMethe** [comment]
FYI: this is the quick fix, I ended up using.

````md
<div class="panel panel-primary">
<div class="panel-heading">
<h4>Group member information</h4>
</div>
<div class="panel-body">

All labs are done in groups, and each group should submit only one of these sheets. Please use the groups assigned on Canvas. Together, decide who will submit the file. This group leader should share their screen, so all group members can view the answers entered. So that all members are given points, make sure to enter the name and student number of all group members present today!

```{r}
# MUST BE PRESENT: the input for the student's name.
lab_name_input(title="Student submitting the file on Canvas", label="Full name", label_id = "UBC student number")
```

<div class="panel panel-default">
<div class="panel-heading">
<h4>Other members present today</h4>
</div>
<div class="panel-body">

In the box below, write the full name and student number of all other group members that are present today. 

<div class="alert alert-danger">
Points will only be given to the student submitting the file and the ones that are listed here. Make sure to list everyone that is present today!
</div>

```{r}
fluidPage(
  fluidRow(column(12, tags$strong('Other group members:'))),
  fluidRow(
    column(6, textInput('zz_group-helped_most_first', label = NULL, placeholder = 'Full name', width = '100%')),
    column(6, textInput('zz_group-helped_most_last', label = NULL, placeholder = 'UBC student number', width = '100%'))),
  fluidRow(
    column(6, textInput('zz_group-helped_most_first', label = NULL, placeholder = 'Full name', width = '100%')),
    column(6, textInput('zz_group-helped_most_last', label = NULL, placeholder = 'UBC student number', width = '100%'))),
  fluidRow(
    column(6, textInput('zz_group-helped_most_first', label = NULL, placeholder = 'Full name', width = '100%')),
    column(6, textInput('zz_group-helped_most_last', label = NULL, placeholder = 'UBC student number', width = '100%')))
  )
```
</div>
</div>

## Remember, this is group work!

**If you don't understand, ask questions to your group members!** Don't be shy to ask for help. These labs are done in group because we expect you to need help from your peers.

**If you find it easy, explain the concepts to your group members!** Teaching others is the best way to integrate knowledge. If you take the time to explain concepts to others, you will understand the material more deeply.

</div>
</div>
````

**Comment by dakep** [comment]
Thanks for sharing your fix and for pointing out the missing error message for individual submissions. This is fixed now in c1863aa.

If you want the student number of all team members than your fix would actually be better. If you want the names and numbers to be saved in the file, though, you would have to change the IDs of the `textInput()` fields from `zz_group-helped_most_first` to something like `zz_group-member_1-name`, `zz_group-member_1-id`, `zz_group-member_2-name`, `zz_group-member_2-id`, …

**Comment by MarieAugerMethe** [comment]
Thanks! Yes, good point! Clearly I used copy and pasted too quickly!
